static/src/javascripts/projects/common/modules/ui/toggles.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+import mediator from 'lib/mediator';
+
+class Toggles {
+    component: ?HTMLElement;
+    controls: Array<HTMLElement>;
+
+    constructor(component: ?HTMLElement = document.body): void {
+        if (component) {
+            const controls = component.querySelectorAll('[data-toggle]');
+
+            this.component = component;
+            this.controls = [...controls];
+
+            mediator.on('module:clickstream:click', clickSpec => {
+                this.reset(clickSpec ? clickSpec.target : null);
+            });
+        }
+    }
+
+    init() {
+        this.controls.forEach(this.prepareControl, this);
+    }
```

Comment:
Is it me or is this `init` method superfluous? Can the implementation be moved into the `constructor`?